### PR TITLE
[clang][analyzer] Improve documentation of checker 'cplusplus.Move' (NFC)

### PR DESCRIPTION
**Pull request by balazske**

### CODE REVIEW

File: clang/docs/analyzer/checkers.rst [hunk 1]
```diff
@@ -420,21 +420,52 @@ around, such as ``std::string_view``.
...
+The checker option ``WarnOn`` controls on what objects the use-after-move is
+checked. The most strict value is ``KnownsOnly``, in this mode only objects are
+checked whose type is known to be move-unsafe. These include most STL objects
+(but excluding move-safe ones) and smart pointers. With option value
+``KnownsAndLocals`` local variables (of any type) are additionally checked. The
+idea behind this is that local variables are usually not tempting to be re-used
+so an use after move is more likely a bug than with member variables. With
+option value ``All`` any use-after move condition is checked on all kinds of
+variables, excluding global variables and known move-safe cases. Default value
+is ``KnownsAndLocals``.
```
Comment (NagyDonat) [inline clang/docs/analyzer/checkers.rst:460]:
```suggestion
The checker option ``WarnOn`` controls on what objects the use-after-move is
checked.
* The most strict value is ``KnownsOnly``, in this mode only objects are
  checked whose type is known to be move-unsafe. These include most STL objects
  (but excluding move-safe ones) and smart pointers.
* With option value ``KnownsAndLocals`` local variables (of any type) are
  additionally checked. The idea behind this is that local variables are
  usually not tempting to be re-used so an use after move is more likely a bug
  than with member variables.
* With option value ``All`` any use-after move condition is checked on all
  kinds of variables, excluding global variables and known move-safe cases.
Default value is ``KnownsAndLocals``.
```
This paragraph was very long, re-flow it into a bullet point list.

**Review by NagyDonat — APPROVED** [review]
LGTM, thanks for updating the docs!

I only have one minor inline remark about formatting.

**Comment by balazske** [comment]
I fixed a test that contained the entire option help description. I think this is not needed, removed it and only included the first line of the description.

**Review by NagyDonat — APPROVED** [review]
The new changes also LGTM, feel free to merge this when you want.

**Pull request merged**